extensions/wikia/AdEngine/js/ml/fmr/fmrInputParser.js

```diff
@@ -0,0 +1,60 @@
+/*global define*/
+define('ext.wikia.adEngine.ml.fmr.fmrInputParser', [
+	'ext.wikia.adEngine.adLogicPageParams',
+	'wikia.log'
+], function (pageParams, log) {
+	'use strict';
+
+	var articleHeightMean = 12169.6328125,
+		logGroup = 'ext.wikia.adEngine.ml.fmr.fmrInputParser';
+
+	function getData() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 35 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)